lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/EvaluatorManager.cs

```diff
@@ -0,0 +1,317 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on 
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+using Org.Apache.REEF.Driver.Evaluator;
+using Org.Apache.REEF.Utilities.Attributes;
+using Org.Apache.REEF.Utilities.Diagnostics;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.Driver
+{
+    /// <summary>
+    /// Manages allocated and failed Evaluators for driver.
+    /// It keeps tracking allocated Evaluators and failed Evaluator ids. Provides methods to 
+    /// add, remove evaluators from the collections so that to provide data for fault tolerant.
+    /// It also tracks master evaluator. 
+    /// </summary>
+    [NotThreadSafe]
+    internal sealed class EvaluatorManager
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(EvaluatorManager));
+        internal const string MasterBatchId = "MasterBatchId";
+        internal const string MapperBatchId = "MapperBatchId";
+
+        private readonly ISet<string> _allocatedEvaluators = new HashSet<string>();
```

Comment:
maybe rename these to `allocatedEvaluatorIDs`, as we don't store `AllocatedEvaluator` instances.